Basic Wayland support

yol opened the pull request:
Pull request for review of #1

Montellese committed:
[settings] split basic functionality of CSettings into CSettingsBase

Montellese committed:
[settings] add abstract CGUIDialogSettingsManagerBase::GetSettingsManager() for easier control over the CSettingsManager being used

garbear committed:
[cosmetic] Whitespace improvements

garbear committed:
Fix signed vs. unsigned warnings

Montellese committed:
[settings] CSettingInt/CSettingString: support translatable and fixed string options

Montellese committed:
[settings] move Save() from CGUIDialogSettingsBase to CGUIDialogSettingsManagerBase

Montellese committed:
[settings] CSettingList: support modifying the list's setting definition

Montellese committed:
[settings] add versioning support

Montellese committed:
[settings] fix saving of updated settings after loading

Montellese committed:
[settings] add support for custom data in CSettingAction

Montellese committed:
[settings] support action settings with custom data

Montellese committed:
[settings] support masking and hiding extensions in CSettingPath

Montellese committed:
[settings] fix FromString/Equals/CheckValidity() for CSettingAddon

Montellese committed:
[settings] support resetting a setting to its default value through the ACTION_DELETE_ITEM action

Montellese committed:
CDateTime: add GetAsDBTime()

Montellese committed:
[settings] add CSettingDate and CSettingTime

Montellese committed:
[settings] add lessthan/lt and greaterthan/gt operators for setting dependencies

Montellese committed:
[settings] support "local" share to not show network shares for path settings

garbear committed:
Joysticks: Add support for mapping game.controller.remote to joystick.xml

garbear committed:
Joysticks: Move IDs to new header file

garbear committed:
Remove unused #include

garbear committed:
JSON-RPC: Fix compiler warnings in VideoLibrary namespace

Montellese committed:
[settings] CSettingsManager: add FindSetting() and InsertSetting() helpers

Montellese committed:
[settings] CSettingsManager: support arbitrary dot separators in setting identifiers

Montellese committed:
[settings] CSettingControlButton: support "file" and "image" formats

garbear committed:
CFile: Remove unused function

Montellese committed:
[settings] make OnSettingAction() available on all setting implementations

Montellese committed:
[settings] add CSettingReference to reference settings

Montellese committed:
[settings] used shared instead of raw pointers

Montellese committed:
[settings] fix label of button controls for CSettingStrings

Montellese committed:
[settings] CSettingsManager: add SetDefault() and SetDefaults()

Montellese committed:
[settings] CSettingsManager: add IsInitialized() and add IsLoaded()

Montellese committed:
[add-ons/settings] move GUIDialogAddonSettings.h/cpp from xbmc/addons/ to xbmc/addons/settings

garbear committed:
Fix compiler warning on Linux

Warning was:

GUITexture.cpp: In member function ‘void CGUITextureBase::Render()’:

GUITexture.cpp:277:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]

   if (y[2] == y[0]) y[2] += 1.0f; if (x[2] == x[0]) x[2] += 1.0f;
   ^~

garbear committed:
VideoPlayer: Fix uninitialized variables

garbear committed:
VideoPlayer: Move #include to cpp file

garbear committed:
Add comment to indicate invalid value

Montellese committed:
[add-ons/settings] migrate add-on settings to settings library

tamland committed:
[addons] refactor model to use ArtMap similar to other media

garbear committed:
Joysticks: Allow keymaps for different controller profiles

garbear committed:
Fix compiler warning in CInputManager

garbear committed:
Joysticks: Add base class for default joystick

garbear committed:
Joysticks: Allow Kodi input from within game

garbear committed:
CGameClient: Fix possible resource leak with open ports

lrusak committed:
[cmake] add FindGLX.cmake and fix system.h defines

lrusak committed:
[cmake] fix FindEGL.cmake defines and adjust system.h

lrusak committed:
windowing/X11: remove unused/broken GLES support

lrusak committed:
windowing/X11: remove uneeded ifdefs

rmrector committed:
Evaluate expressions $EXP for include conditions

rmrector committed:
Evaluate $EXP in other expressions and variable conditions

garbear committed:
Controllers: Move feature logic into CController

garbear committed:
Joysticks: Refactor logic in CKeymapHandler

garbear committed:
Joysticks: Add window ID parameter to lessen dependency on window manager

garbear committed:
Joysticks: Add "holdtime" property to joystick.xml

garbear committed:
Joysticks: Add fallthrough parameter to CDefaultJoystick

garbear committed:
Games: Fix crash due to circular dependency

This also completes a TODO by removing some temporary game code from
CPeripherals.

Rechi committed:
[cleanup] silence unused private field warnings

Rechi committed:
[cleanup] silence unused variable warnings

hudokkow committed:
[TexturePacker] Silence warnings

AlwinEsch committed:
[addons] add callback function to get the MD5 digest of the given text

lrusak committed:
[ios] remove uneeded ifdef HAS_EGL

Montellese committed:
[settings] show asterisks instead of nothing for settings with hidden values

pkerling committed:
Add handler for POSIX SIGTERM and SIGINT signals that shuts the application down

pkerling committed:
Remove superfluous #ifdefs and reorganize includes accordingly

This code is specific to POSIX and in a .cpp file, so the #ifdefs
TARGET_POSIX and __cplusplus are unwarranted.

mkortstiege committed:
[guilib] fadelabel not reading angle tag

Montellese committed:
[settings] refactor CSettingsManager::SetInitialized()

Montellese committed:
[settings] CSettingsManager: support adding a new section after the settings manager has been initialized

Montellese committed:
[settings] CSettingsManager: add necessary locks to AddSection()

Montellese committed:
[settings] CSettingsManager: support adding single settings

Montellese committed:
[add-ons/settings] support adding undefined settings on-the-fly

AlwinEsch committed:
[addons] next binary addon gui interface step

This add the minimum parts to create a window on addon. This still
mostly equal with the Python addon window parts.

During test has this worked, but my experience for GUI things is not the
best and want a feedback.

AlwinEsch committed:
Merge pull request #12208 from AlwinEsch/add-callbacks-4

[addons] next binary addon gui interface step

pkerling commented:
Note: If you want to build it, you will need waylandpp from https://github.com/pkerling/waylandpp until NilsBrause/waylandpp#4 and NilsBrause/waylandpp#5 are merged upstream.

MartijnKaijser committed:
Merge pull request #12207 from pkerling/posix-signals

Stop application on SIGINT and SIGTERM

MartijnKaijser committed:
Merge pull request #12196 from lrusak/opengles-x11-cleanup

X11 Cleanup

MartijnKaijser committed:
Merge pull request #12201 from hudokkow/texturepacker_warnings

[TexturePacker] Silence warnings

MartijnKaijser committed:
Merge pull request #12194 from Rechi/silenceWarnings

[cleanup] silence compiler warnings

pkerling committed:
Fix CAppParamParser::Parse argument type const-ness

A const char*[] is basically a two-dimensional array that only has
const-ness on one level, which means that the contents are not really
const. Replace with const char* const* to have both levels const.
This also removes the need for some const_cast hacks in the startup
code.

pkerling committed:
Remove hackish const_casts

pkerling committed:
Correctly set DEBUG log level for debug builds

The previous code setting LOG_LEVEL_DEBUG did nothing since
CAdvancedSettings::Initialize() sets the level to LOG_LEVEL_NORMAL
anyway. Fix this by overriding the level after possible initialization.

pkerling committed:
Remove log level setting from posix main()

The initial log level is LOG_LEVEL_DEBUG anyway, and setting
LOG_LEVEL_NORMAL with handling the _DEBUG #ifdef override is now handled
in XBMC_Run.

pkerling committed:
Do not use posix main() in WinMain

since it is very confusing for the win32 platform to include code from
platform/posix. It also only saves a few lines.

pkerling committed:
Remove renderGUI var from posix main()

Was never used anyway.

pkerling committed:
Fix posix main() debug detection ifdef from DEBUG to _DEBUG

_DEBUG is set by CMake for debug builds, DEBUG is not.

pkerling committed:
Make WinMain.cpp the CORE_MAIN_SOURCE for the win32 platform

Otherwise, posix/main.cpp would still be used

pkerling committed:
Fix OS X build

HAS_SDL is defined by system.h, so it must be included before the #ifdef

DaveTBlake committed:
Merge scraped album type and label correctly with that derived from tags

notspiff committed:
fix in-tree add-on build environment

FernetMenta commented on cmake/platform/linux/mir.cmake:
```diff
@@ -1,2 +1,2 @@
-set(PLATFORM_REQUIRED_DEPS OpenGl EGL Mir LibDRM)
+set(PLATFORM_REQUIRED_DEPS OpenGl EGL Mir LibDRM Xkbcommon)
```
are you sure you want to touch mir in this early phase?

yol replied:
Changed the xkbcommon lookup so I don't forget about it - mir and wayland both have to convert xkb keysyms to kodi XBMCKey and I copied the map from Mir. We can save it for later though since it will continue to work as-is.

yol replied:
reverted

FernetMenta commented on xbmc/input/keyboard/XkbcommonKeymap.cpp:
```diff
@@ -0,0 +1,348 @@
+ *
+ */
+
+#if defined(HAVE_XKBCOMMON)
```
that type of guards we want to get rid of with cmake. only include the file on the platform it is required
I would put it into windowing, not input. keep in mind that we want to have headless kodi some day. this input folder is quite a mess whereas windowing is already cleaned up

yol replied:
The header should not be a problem, but how do I integrate the .cpp file so that it only compiles on those platforms? Should I use if() like in windowing/CMakeLists.txt?

yol replied:
done

FernetMenta replied:
I guess you found this, right? https://github.com/xbmc/xbmc/pull/12134

yol replied:
Yes, using CORE_PLATFORM_NAME for conditional include now. My only gripe is that if users set CORE_PLATFORM_NAME to e.g. `Wayland` (with a capital `W`), it won't compile since CMake string comparisons are case-sensitive. But maybe I should submit a PR for normalizing the variable in the main Kodi repo.

DaveTBlake committed:
Merge pull request #12216 from DaveTBlake/AlbumTypeMerge

[Fix]Merge album type and label correctly

FernetMenta commented on xbmc/utils/EGLUtils.cpp:
```diff
@@ -0,0 +1,44 @@
+#include "EGLUtils.h"
+#include "log.h"
+
+#if defined(HAVE_LIBEGL)
```
only include file for build with EGL

yol replied:
done

FernetMenta commented on xbmc/utils/StringUtils.h:
```diff
@@ -134,9 +134,87 @@ class StringUtils
-  static std::vector<std::string> Split(const std::string& input, const std::string& delimiter, unsigned int iMaxStrings = 0);
-  static std::vector<std::string> Split(const std::string& input, const char delimiter, size_t iMaxStrings = 0);
-  static std::vector<std::string> Split(const std::string& input, const std::vector<std::string> &delimiters);
+  static std::vector<std::string> Split(const std::string& input, const std::string& delimiter, unsigned int iMaxStrings = 0)
```
why do you put bodies into the header?
how is that related to wayland?

yol replied:
I can put Split() back in the .cpp, but SplitTo() would have to stay because it is a template.
I wanted to use a std::set for the EGL extensions because it is more appropriate than a std::vector. But Split() only returned a std::vector which I would then have to copy to a std::set, which seemed pointless to me.
If you want to avoid the StringUtils change I can revert it and copy the vector to the set after splitting.

FernetMenta replied:
@notspiff what is your opinion on this?

notspiff replied:
templates in headers is a necessary evil. the other is just evil. having container freedom is a valid reason to use a template for sure.

yol replied:
Split() moved

FernetMenta commented on xbmc/windowing/WinEvents.cpp:
```diff
@@ -39,6 +39,10 @@
 #include "android/WinEventsAndroid.h"
 #define WinEventsType CWinEventsAndroid
 
+#elif defined(TARGET_LINUX) && defined(HAVE_WAYLAND)
```
TARGET_LINUX seems superfluous

yol replied:
You're right, removed

FernetMenta commented on xbmc/windowing/wayland/GLContextEGL.cpp:
```diff
@@ -0,0 +1,199 @@
+  EGLint neglconfigs = 0;
+  int major, minor;
+
+  EGLint attribs[] ={
```
no alpha or depth required here?

yol replied:
I admit this is copypaste from Mir. It does seem to work without depth buffer, but we should probably request one anyway. But why do you want alpha?

FernetMenta replied:
I only compared with X11. Not sure if we want alpha :) maybe this is a relict from the times the visual was required to be compatible with SDL.

yol replied:
Depth buffer added - music visualization was indeed looking funny without it :-)
My understanding is that we would need alpha buffer on the surface only if the Kodi window should have transparency, which I think is not the case

FernetMenta commented on xbmc/windowing/wayland/WinEventsWayland.cpp:
```diff
@@ -0,0 +1,92 @@
+  throw std::runtime_error("Not implemented");
+}
+
+// FIXME Why is the return value of this ignored everywhere?
```
MessagePump is supposed to die. In some future version rendering should run on its own thread and this thread can send events to application.

yol replied:
I can put the Wayland message dispatcher on an extra thread and have it only push events to CWinEvents like Mir does. This would make it easier to migrate away from MessagePump when it dies.

FernetMenta replied:
However you like. I just wanted to make sure that you are aware of this.

FernetMenta commented on xbmc/windowing/wayland/WinEventsWayland.cpp:
```diff
@@ -0,0 +1,92 @@
+  // Currently, this is only called by other Wayland-related code which is in turn
+  // called from g_WlDisplay->dispatch_pending() in MessagePump() on the same thread
+  // so no locking is necessary
+  m_queue.emplace(*ev);
```
maybe you can call g_application.OnEvent(event) derectly and drop MessagePump

yol replied:
Can OnEvent() be called from all threads?

FernetMenta replied:
currently not but this can be easily changed by using CApplcationMessenger. that originally was designed to have events processed by the main thread.

FernetMenta commented on xbmc/windowing/wayland/WinEventsWayland.h:
```diff
@@ -0,0 +1,52 @@
+public:
+  virtual std::size_t GetQueueSize() override;
+  virtual bool MessagePump() override;
+  virtual void Notify(const Observable& obs, const ObservableMessage msg) override;
```
what is this?
I don't recomment using Kodi's observable. Its has been abused all over the place and you never know what has hooked into it. the call can even block and go out to addons.

yol replied:
Unused. Removed. It could probably be dropped from IWinEvents, too, none of the other windowing systems use it.

FernetMenta replied:
yes, it's only confusing

hudokkow committed:
Merge pull request #12218 from notspiff/repair_intree_build_env

fix in-tree add-on build environment

pkerling commented:
Just noticed that this does not build with gcc. Will fix.

FernetMenta commented:
good work so far. are you feeling comfortable now with wayland-client?

pkerling commented:
I'm not using libwayland-client. Did you mean waylandpp?

FernetMenta commented:
ahh yes. I noticed there is activity on the repo. great!

AlwinEsch committed:
[addons] add support to new addon gui interface for list items

This changes at the basic support about list items to use on addon
window.

On old system are big memleaks present. This new use now shared pointers
on addon and on kodi.

Also shares Kodi his shared pointer of addon list items as void* to addon,
means one shared_ptr becomes generated with new and contains related shared list
item pointer. On every call back becomes this pointer readed and needed
shared pointer for kodi is present.

The destruct of them pointet shared mem pointer becomes done from Kodi during
destruct of addons list item.

During several tests works it very nice and no memleaks or blockages present.

Montellese committed:
[settings] add helper traits and static methods to CSetting implementations

Montellese committed:
[add-ons/settings] add python setting getters/setters for bool/int/number/string

notspiff committed:
success is not failure

AlwinEsch committed:
Merge pull request #12214 from AlwinEsch/add-callbacks-5

[addons] add support to new addon gui interface for list items

AlwinEsch committed:
Merge pull request #12222 from notspiff/fix_reversed_cond

Fix reversed condition

MartijnKaijser committed:
Merge pull request #12035 from rmrector/expressions_everywhere

[guilib] Evaluate skin expressions everywhere

pkerling committed:
Remove default log setting of DEBUG in debug builds

Turns out that this code was actually just overlooked in an earlier cleanup because it did not work and is not actually supposed to be here anymore.

pkerling commented:
Yes, the author has generally been responsive and merged the PRs I put out in a short time, so using waylandpp should be fine for the time being.

popcornmix committed:
ActiveAE: Fix stereo upmix after ADSP removal

Stereo upmix was broken by:
https://github.com/xbmc/xbmc/pull/11765/commits/fc18b5d3d42eee1a5355c134a076f3b03477178a

Glenn-1990 committed:
[PVR] Fix incrementing play count

ksooo committed:
[addons] Coverity: Fix ID 170202:  Resource leaks  (RESOURCE_LEAK)

ksooo committed:
[addons] Coverity: Fix CID 170198:  Control flow issues  (DEADCODE)

ksooo committed:
[addons] Coverity: Fix CID 170195:  Control flow issues  (DEADCODE)

ksooo committed:
[addons] Coverity: Fix CID 170199: (MIXED_ENUMS)

MartijnKaijser committed:
Merge pull request #12125 from Montellese/addon_settings

[add-ons/settings] migrate add-on settings to settings library

MartijnKaijser committed:
Merge pull request #12226 from Glenn-1990/playcount

[PVR] Fix incrementing play count

AlwinEsch committed:
Merge pull request #12227 from ksooo/addons-coverity-fixes

[addons] Coverity fixes

AlwinEsch committed:
[addons] add all needed addon window interface calls

Next step for binary addon interface rework.

This is nearly the last change on addon Window class. There becomes
all parts added to have needed functionality, also is it with supported
parts now nearly equal with a Python addon window interface.

As last changes on window part comes on next request only the support
for the various control types.

AlwinEsch committed:
[addons] add new CAddonInfo class

This add the class who becomes extended step by step to replace cpluff.

Currently include it only the static functions and table who was before
on other place.

This is done on this commit and request to hold coming request smaller
and prevent parts who not direct related to changes.

As next requests comes the basic read of addon.xml and to go over there.

AlwinEsch committed:
[addons] add few missing functions to listitem interface

This add functions where required on new addon interface to have
same functionality as on old system.

Montellese committed:
[settings] modernize code for C++11

Montellese committed:
[settings] adjust code using settings library to C++11

peak3d committed:
Log a message if ListItem-Type is missing / wrong

popcornmix committed:
fixup! ActiveAE: Fix stereo upmix after ADSP removal

MartijnKaijser committed:
Merge pull request #12215 from pkerling/posix-main-refactor

Posix/Win32 main refactor

FernetMenta committed:
Merge pull request #12233 from popcornmix/fix_stereo

ActiveAE: Fix stereo upmix after ADSP removal

jenkins4kodi committed:
Merge pull request #12224 from AlwinEsch/add-callbacks-6

MartijnKaijser committed:
Merge pull request #12232 from AlwinEsch/add-callbacks-7

[addons] add few missing functions to listitem interface

ObvB committed:
VideoPlayer: fix edl comskip processing

hugegreenbug committed:
Fix for a hangup when viewing a PVR recording and pressing next/previous
 - This fix skips channel up/down events for saved recordings to prevent the hang
 - Bug #17476

peak3d committed:
Merge pull request #12234 from peak3d/playlistplayer

Log a message if ListItem-Type is missing / wrong

jenkins4kodi committed:
Merge pull request #12210 from hugegreenbug/master

fritsch committed:
FFmpeg: Bump to 3.3.1-Leia-Alpha

fritsch committed:
Merge pull request #12241 from fritsch/ffmpeg-bump

FFmpeg: Bump to 3.3.1-Leia-Alpha

tamland committed:
[addons] add banner and clearlogo support

AlwinEsch committed:
[cmake][addons] prevent the use of *.txt files

This prevent the use of text files from addon include dir during install
and define by cmake.

AlwinEsch committed:
Merge pull request #12230 from AlwinEsch/add-addon-info

[addons] add new CAddonInfo class

AlwinEsch committed:
[addons] add first gui control addon class for buttons

This add the first class on new addon interface to edit a button
control.

Hold on this alone to see better how the controls becomes used
on new addon system.

On next request are all controls together.

Here the smallest example to edit a control on new way:
```cpp
// this = The addon window class
// 33 = The on skin.xml used control id
kodi::gui::controls::CButton(this, 33).SetVisible(false);
```

AlwinEsch committed:
[addons] rename addon dialogs to use namespace kodi::gui::dialogs::..

Change to make the different gui parts more clear.

AlwinEsch committed:
[addons] fix CID 170221:  Null pointer dereferences

AlwinEsch committed:
[cmake] fix CMakeLists.txt include paths

Montellese committed:
[settings] CSettingPath: fix CID 170212/170213 by initializing all members

Montellese committed:
[utils] CWeather: fix CID 170211

MartijnKaijser committed:
[android] bump to NDK 14b

Montellese committed:
[peripherals] CPeripherals: fix CID 170205

Montellese committed:
[settings] CGUIDialogContentSettings: fix CID 170204

Montellese committed:
[addons/settings] CAddonSettings: reduce verbosity about loading old setting definitions

Montellese committed:
[addons/settings] CAddonSettings: handle the fact that setting conditions use the index of the option instead of the value for comparison

garbear committed:
Merge pull request #12223 from garbear/fix-circular-dep

Games: Fix crash due to circular dependency

MartijnKaijser committed:
Merge pull request #12245 from Montellese/addon_settings_fallout

[addons/settings] Fix some fallout from the add-on settings PR

MartijnKaijser committed:
Merge pull request #12237 from ObvB/master

VideoPlayer: fix edl comskip processing

MartijnKaijser committed:
Merge pull request #12228 from Montellese/addon_settings_new_python_methods

[add-ons/settings] add python setting getters/setters for bool/int/number/string

peak3d committed:
Improve GUI message handling / gui::smartredraw (#12213)

AlwinEsch committed:
Merge pull request #12240 from AlwinEsch/add-callbacks-7

[addons] add first gui control addon class for buttons

MartijnKaijser committed:
Merge pull request #12247 from MartijnKaijser/droid_ndk14b

[android] bump to NDK 14b

razzeee committed:
Merge pull request #12209 from mkortstiege/fadelabel-angle

[guilib] fadelabel not reading angle tag

AlwinEsch committed:
[addons] move AddonProps to new CAddonInfo class

This add all previous on AddonProps defined values to
CAddonInfo who was added by commits before.

On this change is mostl only an all parts a rename done.

The commit comes in alone to hold next changes smaller
and better for review.

The CAddonInfo becomes on comming changes more and more
changed to allow multiple addon instances.

For this is it later not a static definition in CAddon
and also replace the old cpluff system.

ace20022 committed:
[lang] Add ISO 639-2/T codes to the ISO639 table.

tamland committed:
Merge pull request #12244 from tamland/addons_art

[addons] add support for banner and clearlogo

ksooo committed:
[PVR] Add Info support to recently played channel widget.

garbear committed:
Merge pull request #12239 from garbear/fix-warnings

Compiler warnings on various platforms

garbear committed:
Merge pull request #12180 from garbear/joystick-holdtime

Show OSD from in game by holding start/guide/back for 1 second

ace20022 committed:
[lang] Rename all occurrences of ISO6392T to ISO6392B.

FernetMenta committed:
Merge pull request #12084 from garbear/vp-vars

VideoPlayer: Fix uninitialized variables

micahg committed:
Add an option to pass the ffmpeg cookies http protocol avoption instead of sticking cookie values inside the ffmpeg headers http protocol avoption.

This is preferable for streams that periodically update authentication values (such as a session token) that are stored as cookies. If the cookie values are specified in the headers avoption, it conflicts with the http protocols cookies option, which does get updated properly.

The format is different. The value used with the headers avoption is just the cookie value, whereas, the cookies avoption expects a newline delimited list of Set-Cookie values (eg: everything after ‘Set-Cookie: ‘).

AlwinEsch committed:
Merge pull request #12253 from AlwinEsch/add-addon-info

[addons] move AddonProps to new CAddonInfo class

ksooo committed:
Merge pull request #12254 from ksooo/pvr-channel-widget-info

[PVR] Add Info support to recently played channel widget.

FernetMenta committed:
Merge pull request #12257 from micahg/micah-cookies

Use http cookies AVOption (instead of sticking cookies in the header AVOption)

ace20022 committed:
[lang] Add method ConvertToISO6392T which retrieves the 3-Char ISO 639-2/T code of a given language.

peak3d committed:
Dont force dirtyRegions on parent controls (fixes PR12213 issue)

peak3d committed:
Merge pull request #12255 from peak3d/guidirty

Dont force dirtyRegions on parent controls (fixes PR12213 issue)

AlwinEsch committed:
Merge pull request #12204 from AlwinEsch/add_function

[addons] add callback function to get the MD5 digest of the given text

popcornmix committed:
RBP: Add missing TimingConstants.h includes

MartijnKaijser committed:
Merge pull request #12263 from popcornmix/fixtiminginclude

RBP: Add missing TimingConstants.h includes

ace20022 committed:
[lang] Modernize LangCodeExpander.

Montellese committed:
Merge pull request #12251 from Montellese/settings_cpp11

[settings] move to C++11

Rechi committed:
[utils] Variant: return matching iterators for (begin|end)_(array|map) if type is not array or map

With this change std::distance(...) is also correct if the type isn't an array or map.
Without this change there needs to be a check for isArray respectively isObject before iterating over the elements.

haguenau committed:
Correct spelling of "its"

AlwinEsch committed:
[addons] add next step on CAddonInfo to become independent

This change all values on CAddonInfo to private and access
them over functions.

Main reason for them is to make the class more safe for next changes
where CAddonInfo is no longer in CAddon, but globally as replacement
for cpluff.

Set of this value becomes in future mainly done from the class constructor
and no more by external parts.

Rechi committed:
Merge pull request #12264 from Rechi/variantEmptyArrayMap

[utils] Variant: return matching iterators for (begin|end)_(array|map)

MartijnKaijser committed:
Merge pull request #12266 from haguenau/fix-spelling-its

Correct spelling of "its"

AlwinEsch committed:
[addons] add gui control classes to new addon interface

This change alow addons to edit all used controls on his processed window.

For the moment is this mostly the last request to create the global addon
interface function.

Most further requests are to change addon types iself to new style.

AlwinEsch committed:
Merge pull request #12259 from AlwinEsch/add-callbacks-7

[addons] add gui control classes to new addon interface

AlwinEsch committed:
Merge pull request #12260 from AlwinEsch/add-addon-info

[addons] add next step on CAddonInfo to become independent

ace20022 committed:
[lang] Correct the spelling of the region iso norm.

ace20022 committed:
Merge pull request #12258 from ace20022/isolangcodes

[lang] Add ISO 639-2/T codes.

pkerling committed:
Remove unused Keymap.h

leftover from Wayland cleanup

pkerling committed:
Remove erroneous comment from CApplication

pkerling committed:
Add generic version of StringUtils::Split

When the target container is not an std::vector (e.g. an std::set), the
result of Split() must be copied because always returns a std::vector.
Therefore, add a generic version StringUtils::SplitTo that splits a
string into an output iterator irrespective of the underlying container.

The existing Split() functions are kept for compatibility and
reimplemented using SplitTo().

pkerling committed:
Add EGL utility class CEGLUtils

pkerling committed:
Add POSIX mmap RAII helper

pkerling committed:
Add Wayland windowing implementation

pkerling committed:
Add Wayland windowing implementation to CMake and global headers

pkerling committed:
Fix canwindowed setting to only disable (but not enable) windowed modes

It does not make sense to allow the user to enable windowed mode
when the windowing backend does not support it.

pkerling committed:
Display monitors instead of screens for Wayland

pkerling commented:
Rebased on master, I'll open a new PR to get rid of the extra commits.
